R/tokens.R

```diff
@@ -307,36 +307,27 @@ tokens.corpus <- function(x, ..., include_docvars = TRUE) {
 #' object is a \link{tokens} object, and functions to combine \link{tokens}
 #' objects.
 #' @param x object to be coerced or checked
+#' @param concatenator character between multi-word expressions, default is _ (underscore) character
 #' @return \code{as.tokens} returns a quanteda \link{tokens} object
 #' @export
 #' @rdname as.tokens
-as.tokens <- function(x) {
+as.tokens <- function(x, concatenator = '_') {
     UseMethod("as.tokens")
 }
 
 #' @rdname as.tokens
 #' @export
-as.tokens.list <- function(x) {
+as.tokens.list <- function(x, concatenator = '_') {
     result <- tokens_hash(x)
     attr(result, "what") <- "word"
     attr(result, "ngrams") <- 1L
-    attr(result, "concatenator") <- "_"
+    attr(result, "concatenator") <- concatenator
     attr(result, 'padding') <- FALSE
     class(result)[2] <- "tokenizedTexts"
     return(result)
 }
 
 #' @export
-#' @method as.tokens collocations
```

Comment:
We don't have the use the hashing information, so this is really just `as.tokens(x$collocation)`. But I'm not sure this ever really made sense, because it's completely unclear what a "document" should be when the input is not a list. So let's leave it out.